pkg/proxy/healthcheck/worker.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package healthcheck LoadBalancer Healthcheck responder library for kubernetes network proxies
+package healthcheck // import "k8s.io/kubernetes/pkg/proxy/healthcheck"
+
+import (
+	"sync"
+	"time"
+
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	"github.com/golang/glog"
+)
+
+var once sync.Once
+var healthchecker *proxyHC
+
+func run() {
+	allocatedCh := make(chan bool)
+	go func() {
+		healthchecker = proxyHealthCheckFactory()
```

Comment:
I'm not sure why we really need the once/run/Run/allocatedCh convolution, maybe something like: 
```go
hc := newProxyHealthCheckFactory()
go wait.Until(hc.handlerLoop, time.Second, stopCh)
// pass stopCh wherever
```